### PR TITLE
Tolerations and nodeSelector support for extensions' namespace-metadata

### LINKED ISSUE #10836: linkerd-viz helm chart: not able to set tolerations for namespace-metadata job/pod
### What problem are you trying to solve?

I want to be able to provide tolerations using the helm chart to the namespace-metadata pod run by the job 

### How should the problem be solved?

Add possibility to provide tolerations for namespace-metadata in helm chart

### Any alternatives you've considered?

none I could think of

### How would users interact with this feature?

_No response_

### Would you like to work on this feature?

maybe

### PR DESCRIPTION
**Pull request by pssalman**
Subject
Update Helm Charts

Problem
Unable to start viz, jaeger and multicluster when installed with helm on a kubernetes cluster that only have nodepools with taints configured, where the namespace-metadata job fails to run as the pod is not being assigned to a node.

Solution
Update linkerd-viz, linkerd-jaeger and linkerd-multicluster helm charts to support tolerations and nodeSelector

Fixes #10836

Signed-off-by: Anton Salman <anton.salman@gmail.com>

### CODE REVIEW

File: jaeger/charts/linkerd-jaeger/Chart.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ kubeVersion: ">=1.21.0-0"
 name: linkerd-jaeger
 sources:
 - https://github.com/linkerd/linkerd2/
-version: 30.9.1-edge
+version: 30.9.2-edge
```
Comment (mateiidavid) [inline jaeger/charts/linkerd-jaeger/Chart.yaml:14]: We should restore this change. We tend to update charts as part of releases.

File: viz/charts/linkerd-viz/Chart.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ kubeVersion: ">=1.21.0-0"
 name: "linkerd-viz"
 sources:
 - https://github.com/linkerd/linkerd2/
-version: 30.9.1-edge
+version: 30.9.2-edge
```
Comment (mateiidavid) [inline viz/charts/linkerd-viz/Chart.yaml:14]: same here, this should be kept out of the change

File: jaeger/charts/linkerd-jaeger/Chart.yaml (hunk 1 above)
Comment (pssalman) [reply jaeger/charts/linkerd-jaeger/Chart.yaml]: Reverted back the change as well as on the README file

File: viz/charts/linkerd-viz/Chart.yaml (hunk 2 above)
Comment (pssalman) [reply viz/charts/linkerd-viz/Chart.yaml]: Reverted back the change as well as on the README file

Comment (pssalman) [reply viz/charts/linkerd-viz/Chart.yaml]: I have also reverted the multicluster helm chart version.

File: multicluster/charts/linkerd-multicluster/values.yaml [hunk 3]
```diff
@@ -74,6 +74,11 @@ namespaceMetadata:
     # @default -- imagePullPolicy
     pullPolicy: ""
 
+  # -- Node selectors for the gateway pod
+  nodeSelector: {}
+  # -- Tolerations for the gateway pod
+  tolerations: []
```
[the remark this answers is not in the document]
Comment (pssalman) [reply multicluster/charts/linkerd-multicluster/values.yaml]: Updated

**Review by alpeb — COMMENTED** [review]
You still need to call `./bin/helm-docs` to have the charts' README.md files be automatically updated :wink:

**Comment by pssalman** [comment]
> You still need to call `./bin/helm-docs` to have the charts' README.md files be automatically updated 😉

Seems I had a slightly different name between the comments in values.yaml and the README.md file. I just fixed that.
Thanks for checking it out.

**Review by alpeb — APPROVED** [review]
Thanks @pssalman , this looks good to me :+1:

**Review by adleong — APPROVED** [review]

**Pull request merged**